Update 'gnatsd' + 'Dockerfile'.

bmcustodio opened the pull request:
I've split this in two separate commits for better analysis.

---

> (6a60a2c) Update 'gnatsd' to v1.0.2.
> This updates the bundled version of 'gnatsd' to v1.0.2. I've also
> rebuilt 'route_checker' using Go 1.9.0, and updated 'README.md' and the
> deployment descriptors accordingly.

The official image's `gnatsd` binary is now made available at https://github.com/nats-io/nats-docker, so I think we can use that instead of building our own. I've nonetheless kept instructions for building, but made them match https://github.com/nats-io/gnatsd/blob/v1.0.2/Dockerfile#L8.

---

(707b35e) Update 'Dockerfile'.
> Bumped to 'alpine:3.6'. Replaced the 'sudo' call in 'run.sh' with
> 'exec', which maks 'gnatsd' PID 1 which in turn allows for better
> signal handling (the container won't be forcefully killed anymore
> allowing for NATS to exit cleanly).

I made a few changes to `Dockerfile`:

* Bumped to `alpine:3.6`.
* Removed deprecated `MAINTAINER` instruction in favor of a now-recommended `LABEL` one.
* Removed `sudo` as a dependency (more on this later) and added a manual clean of `/var/cache/apk` since neither `--no-cache` nor `apk cache clean` work as one would expect when `apk update` is used.

I also made a few changes to `run.sh`:

* Removed the creation of the `nats` user and `sudo` group. I believe its safe to run `gnatsd` as `root` inside the container and this way we can make things simpler.
* Use `exec /gnatsd` instead of `sudo -E -u nats /gnatsd`. This will make `gnatsd` run as PID `1` and allow for proper signal handling (container stopping using `Ctrl+C` or `docker stop`, etc.).

(This also helped shave a couple of MB from the image, although it probably doesn't matter much.)

pires commented on deployment-tls.yaml:
```diff
@@ -12,7 +12,7 @@ spec:
       containers:
       - name: nats
-        image: quay.io/pires/docker-nats:0.9.6_1
+        image: quay.io/pires/docker-nats:1.0.2_1
```
Remove `_1` as I usually use that as patch version.

pires commented on deployment.yaml:
```diff
@@ -12,7 +12,7 @@ spec:
       containers:
       - name: nats
-        image: quay.io/pires/docker-nats:0.9.6_1
+        image: quay.io/pires/docker-nats:1.0.2_1
```
Remove `_1` as I usually use that as patch version.

pires commented on README.md:
````diff
@@ -11,36 +11,46 @@ NATS cluster on top of Kubernetes made easy.
 One must change `deployment.yaml` accordingly, commit everything and proceed to push a new tag that will trigger an automatic build:
 ```
-git tag 0.9.6_1
+git tag 1.0.2_1
````
Remove `_1` as I usually use that as patch version.

bmcustodio reviewed (commented):
@pires done! Do you want me to squash these commits?

pires reviewed (changes requested):
Please squash.

bmcustodio committed:
Update 'gnatsd' to v1.0.2.

This updates the bundled version of 'gnatsd' to v1.0.2. I've also
rebuilt 'route_checker' using Go 1.9.0 and updated 'README.md' and the
deployment descriptors accordingly.

I've also bumped to 'alpine:3.6', and replaced the 'sudo' call in
'run.sh' with 'exec', which makes 'gnatsd' PID 1, which in turn allows
for better signal handling (the container won't be forcefully killed
anymore, allowing for NATS to exit cleanly).

bmcustodio commented:
@pires done!
